src/nupic/math/ArrayAlgo.hpp

```diff
@@ -3779,7 +3779,7 @@ namespace nupic {
       // n1 is the number floats we can process in parallel with xmm
       // n2 is the number of "stragglers" what we will have to do one by one ( < 4)
       nupic::Real32 count = 0;
-      long x_addr = (long) begin; // 8 bytes on 64 bits platforms
+      long long x_addr = (long long) begin; // 8 bytes on 64 bits platforms
```

Comment:
@rcrowder - Your change fixes Windows but breaks Linux. A `long` is 32 bit on Windows and 64 bit on nearly any other 64 bit OS. Please consider the impact on other platforms rather than just fixing it on Windows :)

My recommendation for this case is to see if you can use `nupic::Handle` (which is `void *`) as the type of `x_addr`. That will be 32 bits on 32 bit platforms and 64 on 64 bit platforms. If that doesn't allow you to do math on it then we will probably have to use a preprocessor conditional and choose UInt64 or UInt32 based on the platform.